api/client/__init__.py

```diff
@@ -328,3 +329,35 @@ def get_descendant_regions(self, region_id, descendant_level=None,
         """
         return lib.get_descendant_regions(self.access_token, self.api_host, region_id,
                                           descendant_level, include_historical, include_details)
+
+
+    def get_available_timefrequency(self, **selection):
+        """Given a selection, return a list of frequencies and time ranges.
+        The results are ordered by coverage-optimized ranking.
+
+        Parameters
+        ----------
+        metric_id : integer, optional
+        item_id : integer, optional
+        region_id : integer, optional
+        partner_region_id : integer, optional
+
+        Returns
+        -------
+        list of dicts
+
+            Example::
+
+                 [{
+                    'startDate': '2000-02-18T00:00:00.000Z',
+                    'frequencyId': 3,
+                     'endDate': '2020-03-12T00:00:00.000Z',
+                     'name': '8-day'
```

Comment:
I noticed that we have an extra "period" field returned from API, could be removed though, viz is not using that neither